netty/src/main/java/io/grpc/netty/WriteQueue.java

```diff
@@ -106,7 +112,14 @@ ChannelFuture enqueue(Object command, boolean flush) {
    *              enqueue will schedule the flush.
    */
   ChannelFuture enqueue(Object command, ChannelPromise promise, boolean flush) {
-    queue.add(new QueuedCommand(command, promise));
+    final QueuedCommand queuedCommand;
+    if (command instanceof QueuedCommand) {
+      queuedCommand = (QueuedCommand) command;
+      queuedCommand.promise(promise);
+    } else {
+      queuedCommand = new InternalQueuedCommand(command, promise);
```

Comment:
Are there any commands that are not QueuedCommands?  Would it be easier to just forcefully cast  and not worry about other object writes?